### PR TITLE
union: fix union dual table order by

### PR DESCRIPTION
**Pull request by lysu**
ref via https://github.com/pingcap/tidb/issues/8201
test by https://github.com/pingcap/tidb-test/pull/645

```
select 2 as a from dual union select 1 as b from dual order by a 
```

need order by whole union resut, but current version `order by` is attached to last dual.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/parser/22)

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
And I think we'd better submit a PR in TiDB repo to add an integration test,
which can be merged with this PR together.

File: go.sum1 [hunk 1]
```diff
@@ -134,13 +134,16 @@ github.com/pingcap/goleveldb v0.0.0-20171020122428-b9ff6c35079e h1:P73/4dPCL96rG
 github.com/pingcap/goleveldb v0.0.0-20171020122428-b9ff6c35079e/go.mod h1:O17XtbryoCJhkKGbT62+L2OlrniwqiGLSqrmdHCMzZw=
 github.com/pingcap/kvproto v0.0.0-20181028030329-855d2192cdc7 h1:CYssSnPvf90ZSbFdZpsZGSI7y+drG1EfKxqTOnKnHb0=
 github.com/pingcap/kvproto v0.0.0-20181028030329-855d2192cdc7/go.mod h1:0gwbe1F2iBIjuQ9AH0DbQhL+Dpr5GofU8fgYyXk+ykk=
+github.com/pingcap/kvproto v0.0.0-20181105061835-1b5d69cd1d26/go.mod h1:0gwbe1F2iBIjuQ9AH0DbQhL+Dpr5GofU8fgYyXk+ykk=
```
Comment (shenli) [inline go.sum1]: Why this is changed?

**Comment by zz-jason** [comment]
@XuHuaiyu @lysu, For integration test, It's better to add it to the tidb-test repo, not the tidb repo.

**Comment by zz-jason** [comment]
@lysu Please add some UT

File: go.sum1 (hunk 1 above)
Comment (lysu) [reply go.sum1]: because tidb version is changed, this indirect change is caused by tidb

**Comment by lysu** [comment]
@XuHuaiyu @zz-jason  PTAL

**Pull request merged**